### PR TITLE
feat(next): `createSSGHelpers` to `createServerSideHelpers`

### LINKED ISSUE #4041: feat: rename `createSSGHelpers`
### Describe the feature you'd like to request

They were never designed only to work with `getStaticProps`. We should have better name for it.

### Describe the solution you'd like to see

- Rename
- Alias to old name with `@deprecated`-notice
- Update docs, especially https://trpc.io/docs/nextjs/ssr that don't mention gSSP at all

### Describe alternate solutions

no

### Additional information

- The new name doesn't have to have `Proxy` in it (that was just to not conflict with interop stuff)
- Don't touch anything in any `**/interop/**`-file

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR implementing this feature!

### PR DESCRIPTION
**Pull request by mozzius**
Closes #4041

## 🎯 Changes

Renamed `createSSGHelpers` to `createServerSideHelpers`. Aliased the old name to the new one.
Edited the docs to use the new name.
Changed the examples to use the new name.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: packages/react-query/src/ssg/index.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 
 export {
-  createSSGHelpers,
+  createServerSideHelpers,
   /**
-   * @deprecated - use `createSSGHelpers` instead
+   * @deprecated - use `createServerSideHelpers` instead
    */
-  createSSGHelpers as createProxySSGHelpers,
+  createServerSideHelpers as createProxySSGHelpers,
```
Comment (KATT) [inline packages/react-query/src/ssg/index.ts:8]: plz keep a a test that uses `createProxySSGHelpers`

Comment (KATT) [reply packages/react-query/src/ssg/index.ts:8]: or maybe copy/paste the old unchanged files to a `./deprecated`-folder

Comment (mozzius) [reply packages/react-query/src/ssg/index.ts:8]: Ah I thought of that, but accidentally did it only in the dehydration test. I'll move both to a `./deprecated` folder, I think that sounds neatest.

File: packages/tests/server/react/dehydrate.test.tsx [hunk 2]
```diff
@@ -24,40 +21,9 @@ test('dehydrate', async () => {
...
-      "title": "first post",
-    }
-  `);
-});
-
-test('deprecated alias', async () => {
-  const { db, appRouter } = factory;
-  const ssg = createProxySSGHelpers({ router: appRouter, ctx: {} });
+  // typescript doesn't know that it definitely has 2 elements in the array
+  if (!cache || !cache2) throw Error("can't happen");
```
Comment (mozzius) [inline packages/tests/server/react/dehydrate.test.tsx:25]: I was getting a type error here, presumably due to `noUncheckedIndexedAccess`? This gets rid of the error but it's terrible, is there a better way of asserting these definitely aren't undefined?

**Review by KATT — COMMENTED** [review]
Oh I just noticed that you targeted `next`, we could do this on the `main` branch!

**Comment by mozzius** [comment]
Ah ok. I did it on `next` because @juliusmarminge mentioned that there had already been some renaming work done (to remove "proxy")

**Comment by KATT** [comment]
> Ah ok. I did it on `next` because @juliusmarminge mentioned that there had already been some renaming work done (to remove "proxy")

Ahhhh yeah, will cause some conflicts, but I still think we should do this on `main` :)

**Comment by juliusmarminge** [comment]
> Ah ok. I did it on `next` because @juliusmarminge mentioned that there had already been some renaming work done (to remove "proxy")

Yea we can still add this alias here - just gotta be sure to update it on next so it's aliasing correctly on there

**Comment by mozzius** [comment]
Oh man I tried rebasing but it did not work 😬 gonna close this one and just start fresh

**Pull request closed without merging**